src/main/java/org/jenkinsci/plugins/workflow/multibranch/BranchJobProperty.java

```diff
@@ -67,23 +66,25 @@ synchronized void setBranch(@Nonnull Branch branch) {
             @Override public boolean hasPermission(Authentication a, Permission permission) {
                 // This project is managed by its parent and may not be directly configured or deleted by users.
                 // Note that Item.EXTENDED_READ may still be granted, so you can still see Snippet Generator, etc.
-                if (branch instanceof Branch.Dead && (permission == Job.BUILD)) {
-                    // if dead, nobody, not even SYSTEM may build this job
-                    return false;
```

Comment:
Only effective against code paths which do a permission check; for example, `scheduleBuild` overloads (called, e.g., from `TimerTrigger`) do not. `isBuildable` is checked more broadly.